bot.py

```diff
@@ -40,7 +37,316 @@ async def hello(ctx):
+                roles.append(role)
+            else:
+                await ctx.send(f"Could not find `{role_name}` role. Check spelling.")
+                return
+        roles = [role for role in roles if role is not None]
+        if not roles:
+            await ctx.send("Could not find one or more roles. Check spelling.")
+            return
+
+    members = set()
+    if " and " in output_string:
+        role_names = " and ".join(role.name for role in roles)
+        for role in roles:
+            if not members:  # start if members is empty
+                members.update(role.members)
+            else:
+                members &= set(role.members)
+    elif " or " in output_string:
+        role_names = " or ".join(role.name for role in roles)
+        for role in roles:
+            if not members:  # start if members is empty
+                members.update(role.members)
+            else:
+                members &= set(role.members)
+    else:
+        role_names = role_names[0]
+        members = set(ctx.guild.members)
+
+    for not_role in not_roles:
+        members -= set(not_role.members)
+
+    members = list(members)
+    member_names = ", ".join(member.mention for member in members)
+    not_role_names = ", ".join(not_role.name for not_role in not_roles)
+    if not_roles:
+        if only_nots_in_str is True:
+            await ctx.send(
+                f"The members without roles `{not_role_names}` are: {member_names}."
+            )
+        else:
+            await ctx.send(
+                f"The members with the roles `{role_names}`"
+                f" but not the roles `{not_role_names}` are: {member_names}."
+            )
+    else:
+        await ctx.send(
+            f"The members with the roles `{role_names}` are: {member_names}."
+        )
+
+
+def converting_string(input_string):
+    """Docstring placeholder"""
+    output_string = ""
+    start_index = 0
+    end_index = 0
+    while start_index < len(input_string):
+        start_index = input_string.find("{", end_index)
+        if start_index == -1:
+            output_string += input_string[end_index:]
+            break
+        output_string += input_string[end_index:start_index]
+        end_index = input_string.find("}", start_index)
+        if end_index == -1:
+            output_string += input_string[start_index:]
+            break
+        function_string = input_string[start_index + 1 : end_index]  # noqa: E203
+        if function_string.startswith("list_members "):
+            second_string = function_string[13:]
+            output_string += second_string
+        elif function_string.startswith("count_members "):
+            second_string = function_string[14:]
+            output_string += second_string
+        else:
+            output_string += "{" + function_string + "}"
+        end_index += 1
+    return output_string
+
+
+@bot.command()
+async def searching_for_roles(ctx, output_string):
+    """Docstring placeholder"""
```

Comment:
Co robi ta funkcja?   
Na pierwszy rzut oka pomyślałbym, że to ona ma filtrować listę użytkowników, ale widzę tu kilka rzeczy, które świadczą inaczej. (położenie w kodzie, jest to komenda dla bota, przyjmuje dziwne argumenty...)

Reply:
jest to w prosty sposób zmieniona funkcja list_members. Zmieniona w taki sposób, żeby zamiast przyjmować argument z czatu discorda, argumentem powinna być przekonwertowana wiadomość z poprzedniej funkcji, czyli converting_string. Pierwszy argument, czyli ctx, z tego co wiem jest potrzebny, jeżeli chce się wyciągać informacje z serwera. Dodatkowo, zmieniłem zwracanie argumentów, żeby zamiast wiadomości końcowe były wysyłane do czatu, powinny być zwracane wewnątrz kodu, tam gdzie będę potem chciał.

Reply:
Zamykam, bo widzę, że już jej nie ma